sale_blanket_order/models/planned_blanket_orders.py

```diff
@@ -0,0 +1,73 @@
+# coding: utf-8
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class BlanketPlannedOrder(models.Model):
+    _name = 'sale.blanket.planned.order'
+    _description = 'Blanket Planned Order'
+    blanket_order_id = fields.Many2one(
+        'sale.blanket.order', required=True, ondelete='cascade')
+    next_order_date = fields.Date(string='next order date', required=True)
+    line_ids = fields.One2many(
+        'sale.blanket.planned.order.line', 'planned_order_id', string='Order lines',
+        copy=True)
+
+    @api.multi
+    def create_order(self):
+        for rec in self:
+            if fields.Date.today() == rec.next_order_date:
```

Comment:
Better to use filtered.

    for rec self.filtered(lambda r: r.next_order_date == fields.Date.today()):

Reply:
fixed!